service/processor/src/test/java/com/google/auto/service/processor/AutoServiceProcessorTest.java

```diff
@@ -85,4 +85,63 @@ public void badMultiService() {
     assertThat(compilation).failed();
     assertThat(compilation).hadErrorContaining(MISSING_SERVICES_ERROR);
   }
+
+  @Test
+  public void generic() {
+    Compilation compilation =
+        Compiler.javac()
+            .withProcessors(new AutoServiceProcessor())
+            .compile(
+                JavaFileObjects.forResource("test/GenericService.java"),
+                JavaFileObjects.forResource("test/GenericServiceProvider.java"));
+    assertThat(compilation).succeededWithoutWarnings();
+    assertThat(compilation)
+        .generatedFile(StandardLocation.CLASS_OUTPUT, "META-INF/services/test.GenericService")
+        .contentsAsUtf8String()
+        .isEqualTo("test.GenericServiceProvider\n");
+  }
+
+  @Test
+  public void genericWithVerifyOption() {
+    Compilation compilation =
+        Compiler.javac()
+            .withProcessors(new AutoServiceProcessor())
+            .withOptions("-Averify=true")
+            .compile(
+                JavaFileObjects.forResource("test/GenericService.java"),
+                JavaFileObjects.forResource("test/GenericServiceProvider.java"));
+    assertThat(compilation).succeeded();
+    assertThat(compilation)
+        .hadWarningContaining(
+            "Service provider test.GenericService is generic, so it can't be named exactly by"
```

Comment:
This makes no sense: it should read `"Service provider test.GenericServiceProvider"`. In my test application, this actually becomes

````
Service provider test.GenericService is generic...
Service provider test.GenericService is generic...
Service provider test.GenericService is generic...
Service provider test.GenericService is generic...
...
````

instead of the more explicit

````
Service provider test.FirstGenericService is generic...
Service provider test.SecondGenericService is generic...
Service provider test.ThirdGenericService is generic...
Service provider test.FourthGenericService is generic...
...
```

Reply:
Don't you have an indication of where the error is coming from? It should indicate `test/GenericService.class`.

Reply:
```
@AutoService(GenericService.class)
class NonGenericServiceProvider implements GenericService<String>
```

Why do you say it makes no sense to add `@SuppressWarnings("rawtypes")` here? You are using the raw type `GenericService` in the `@AutoService` annotation. Of course you don't have any choice in the matter, since there's no such thing as `GenericService<String>.class`, but I still think it's reasonable for the specific `@SuppressWarnings` to be `rawtypes`.

Reply:
> Don't you have an indication of where the error is coming from? It should indicate test/GenericService.class.

No, I don't. I'm using AdoptOpenJDK v15 at the moment. My build tool is Gradle. I'd have to make MCVE to show you, I probably will in the coming days. Is it possible that it's because my service providers are generated?

> Why do you say it makes no sense to add `@SuppressWarnings("rawtypes")` here? You are using the raw type `GenericService` in the `@AutoService` annotation. Of course you don't have any choice in the matter, since there's no such thing as `GenericService<String>.class`, but I still think it's reasonable for the specific `@SuppressWarnings` to be `rawtypes`.

Thanks for the explanation. It's just that it's *unexpected*, especially with the error message that doesn't reference GenericServiceProvider at all.

Reply:
> > Don't you have an indication of where the error is coming from? It should indicate test/GenericService.class.
> 
> No, I don't. I'm using AdoptOpenJDK v15 at the moment. My build tool is Gradle. I'd have to make MCVE to show you, I probably will in the coming days. Is it possible that it's because my service providers are generated?
> 
Actually, I see it, it's more clear now. My bad. The issue was that I already had an implementation for `warning(String, Object...)` in my version and I didn't copy your version of `warning(String,Element,AnnotationMirror)` because I had no compilation issue after copying the initial issue.

The message is akin to

````
D:\...\GenericServiceImplementor.java:7: warning: Service provider test.GenericService is generic, so it can't be named exactly by @AutoService. If this is OK, add @SuppressWarnings("rawtypes").
@Generated(
^
````

This, combined with the initial misunderstanding of the class make that indeed the code is OK and needs no further adaptation. I'm really sorry for taking your time for my sillinesses.

Reply:
No problem! I really appreciate your trying it out so quickly.